crates/hir/src/diagnostics.rs

```diff
@@ -310,3 +311,10 @@ pub struct TraitImplMissingAssocItems {
     pub impl_: AstPtr<ast::Impl>,
     pub missing: Vec<(Name, AssocItem)>,
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct TraitImplReduntantAssocItems {
```

Comment:
```suggestion
pub struct TraitImplRedundantAssocItems {
```
and where else you wrote retuntant